Shopify/cli
# Encourage scope granting in GraphiQL

### WHY are these changes introduced?

When a developer changes the access scopes for their app, and wants to continue to prototype and build, they almost certainly will need the merchant of their dev store -- themselves! -- to grant said access. For the app home, a refresh will do the job with declarative scopes. But lots of building now happens in GraphiQL, and this doesn't have the same behaviour. 



### WHAT is this pull request doing?

Firstly the token used by GraphiQL is refreshed often now, so its scopes should always be up to date with those granted by the dev store.

This PR adds a couple of helpful nudges:
- if the access scopes in the in-use app toml don't match those that GraphiQL's access token is attempting to use, this is a sign that the developer has changed them. we show a message in the top-bar of GraphiQL encouraging the dev to run `shopify app deploy`
- after running `shopify app deploy`, if the developer has changed their access scopes within their newly released app version, we show a next step about granting these new access scopes in their dev store



<img width="600" alt="image" src="https://github.com/Shopify/cli/assets/234027/0e6fb18b-4af9-4c82-be03-f8d0829f706d">
<img width="1800" alt="image" src="https://github.com/Shopify/cli/assets/234027/8f5b338d-b2e0-475e-b695-b69635019468">


### How to test your changes?



1. Run `shopify app dev`, and launch GraphiQL
2. Leaving dev running, go to your app toml, and add a new scope, save the file
3. The banner shown above should appear. If you change the scopes back, it will disappear.
4. With changed scopes, run `shopify app deploy`. Choose to create and release a new version
5. The confirmation message should mention going to your dev store, as per the above

### Post-release steps



n/a

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
